### PR TITLE
Spike/run script

### PR DESCRIPTION
**Pull request by connoratrug**
Initial setup to run script as package based on https://gist.github.com/SietsmaRJ/fd4ecba11f55a34db29991d578c10ca8

- Put gist script into package
- Rename package to comply with r naming best practises 
- Split off large functions and partial apply to pass data from global scope 
- Remove hard coded file reading from main function and move to separate demo function

### CODE REVIEW

**Review by marikaris — COMMENTED** [review]
ERROR: dependency ‘gridExtra’ is not available for package ‘mosaicr’

File: R/mozaiekberekentool.R [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+#' @import grDevices
+#' @import grid
+#' @import gridExtra
+#' @importFrom utils read.table setTxtProgressBar txtProgressBar
+#' @importFrom graphics plot text
+#' @export
+#'
+#Versionnumber: 0.6.0.0
+MosaicCalculator <- function(exp.nr, gender, snpm.data, deviations, outfile){
+  print("Dank u voor het kiezen van Mozaiek Bereken Tool, ontwikkeld door Robert Sietsma")
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: Please remove this

File: R/mozaiekberekentool.R [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  #Set the option of the output decimal to comma
+  options(OutDec = ",")
+  
+  cat("Ombouwen van deviations naar bruikbare parameters.","\n")
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: Can you translate this?

File: R/mozaiekberekentool.R [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+#' @import gridExtra
+#' @importFrom utils read.table setTxtProgressBar txtProgressBar
+#' @importFrom graphics plot text
+#' @export
+#'
+#Versionnumber: 0.6.0.0
+MosaicCalculator <- function(exp.nr, gender, snpm.data, deviations, outfile){
+  print("Dank u voor het kiezen van Mozaiek Bereken Tool, ontwikkeld door Robert Sietsma")
+  #Set the option of the output decimal to comma
+  options(OutDec = ",")
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: I think for international purposes we want to use a . as separator for decimals?

File: R/mozaiekberekentool.R [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  
+  cat("Ombouwen van deviations naar bruikbare parameters.","\n")
+  eventsoutput <- process.deviation.line(deviations, gender)
+
+  cat("Berekenen van kwaliteit aspecten array.","\n")
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: Can you translate this?

File: R/mozaiekberekentool.R [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  eventsoutput[with(eventsoutput, Event %in% c("LOH", "Allelic Imbalance") & Length <= 5000000),] <- NA
+  eventsoutput <- eventsoutput[complete.cases(eventsoutput),]
+  eventsoutput$Length <- NULL
+  eventsoutput$Probes <- NULL
+
+  #Returning the number of rows that are about to be used in the apply calculation
+  aantalevents <- nrow(eventsoutput)
+  line1 <- paste(aantalevents, " events die groter zijn dan 150kb, meer dan 10 probes hebben en LOH/AO gebieden groter dan 5Mb.", sep = "")
+  cat(line1, "\n")
+  cat("Bezig met het filteren en berekenen van de overgebleven events, een ogenblik geduld alstublieft...","\n")
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: Can you translate this?

File: R/mozaiekberekentool.R [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+  eventsoutput[eventsoutput$Length <= 100000,] <- NA
+  eventsoutput <- eventsoutput[complete.cases(eventsoutput),]
+  eventsoutput[with(eventsoutput, Event %in% c("LOH", "Allelic Imbalance") & Length <= 5000000),] <- NA
+  eventsoutput <- eventsoutput[complete.cases(eventsoutput),]
+  eventsoutput$Length <- NULL
+  eventsoutput$Probes <- NULL
+
+  #Returning the number of rows that are about to be used in the apply calculation
+  aantalevents <- nrow(eventsoutput)
+  line1 <- paste(aantalevents, " events die groter zijn dan 150kb, meer dan 10 probes hebben en LOH/AO gebieden groter dan 5Mb.", sep = "")
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: Can you translate this?

File: R/mozaiekberekentool.R [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+  result.lines <- c(result.lines, "RESULTS:")
+  result.lines <- c(result.lines, "")
+  
+  #Returning the number of rows that are about to be used in the apply calculation
+  result.lines <- c(result.lines, paste(aantalevents, "events that a larger then 150kb."))
+  result.lines <- c(result.lines, "More then 10 probes have a LOH/AO area larger then 5Mb.")
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("SNPs buiten afwijkingen: ", SNPs_used_in_averageBAF))
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("Gemiddelde BAF: ", meanaverageBAF))
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: Can you translate this?

File: R/mozaiekberekentool.R [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+  #Returning the common array values
+  result.lines <- NULL
+  result.lines <- c(result.lines, "RESULTS:")
+  result.lines <- c(result.lines, "")
+  
+  #Returning the number of rows that are about to be used in the apply calculation
+  result.lines <- c(result.lines, paste(aantalevents, "events that a larger then 150kb."))
+  result.lines <- c(result.lines, "More then 10 probes have a LOH/AO area larger then 5Mb.")
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("SNPs buiten afwijkingen: ", SNPs_used_in_averageBAF))
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: Can you translate this?

File: R/mozaiekberekentool.R [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("Gemiddelde BAF: ", meanaverageBAF))
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("STDev BAF: ", sdaverageBAF))
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("Gemiddelde afwijking van BAF 0.5: ", MADaverageBAF))
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: Can you translate this?

File: R/mozaiekberekentool.R [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+  result.lines <- c(result.lines, "")
+  result.lines <- c(result.lines, c("Correctiefactor: ", correctionfactor))
```
Comment (marikaris) [inline R/mozaiekberekentool.R]: Can you translate this?

**Review by marikaris — CHANGES_REQUESTED** [review]
Please rename the "mozaiekberekentool.R" to something english

Please remove the "square" file

**Comment by connoratrug** [comment]
> ERROR: dependency ‘gridExtra’ is not available for package ‘mosaicr’

Please install it, or use devtools install all (```devtools::install_deps(dependencies = TRUE)```)

File: R/mozaiekberekentool.R (hunk 1 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: done

File: R/mozaiekberekentool.R (hunk 3 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: I don't know, i suggest we check with Marloes and Lennart, i suggest to leave is as is for now until we know what the users would like, i also do not know is the string processing in the script depends of the output decimal being an comma.

File: R/mozaiekberekentool.R (hunk 2 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: done

File: R/mozaiekberekentool.R (hunk 4 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: done

File: R/mozaiekberekentool.R (hunk 6 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: done

File: R/mozaiekberekentool.R (hunk 8 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: done

File: R/mozaiekberekentool.R (hunk 10 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: done

File: R/mozaiekberekentool.R (hunk 9 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: done

File: R/mozaiekberekentool.R (hunk 5 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: done

File: R/mozaiekberekentool.R (hunk 7 above)
Comment (connoratrug) [reply R/mozaiekberekentool.R]: done

File: R/mozaiekberekentool.R (hunk 3 above)
Comment (marikaris) [reply R/mozaiekberekentool.R]: Agree

File: R/events.R [hunk 11]
```diff
@@ -0,0 +1,477 @@
...
+      CNVType <- "CN Loss"
+    }
+  }
+
+  #Making of Output Chromosome Region (OCR)
+  OCR <- paste("chr",chr,":",formatC(start, big.mark = "."),"-",formatC(stop, big.mark = "."), sep = "")
+
+  #Homozygous copy losses disturb the BAF signal
+  if(CNVType == "Homozygous Copy Loss"){
+    return(c(OCR,"Homozygote deletion","NA","NA","NA","NA","NA","NA","NA","NA"))
```
Comment (marikaris) [inline R/events.R]: "Homozygote" is still Dutch, should be "Homozygous"

**Review by marikaris — CHANGES_REQUESTED** [review]

File: R/events.R (hunk 11 above)
Comment (connoratrug) [reply R/events.R]: done

**Review by marikaris — APPROVED** [review]

**Pull request merged**